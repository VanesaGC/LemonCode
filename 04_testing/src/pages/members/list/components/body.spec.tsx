import * as React from "react";
import { shallow } from "enzyme";
import { Body } from "./body";

describe("pagesmemberslistcomponents (body) tests", () => {
  it("should render as expected when passing required properties", () => {
    // Arrange
    const props = {
      members: [
        {
          id: 1,
          name: "Mar√≠a",
          avatarUrl: "maria.png"
        },
        {
          id: 2,
          name: "Pedro",
          avatarUrl: "pedro.png"
        }
      ]
    };
    // Act
    const component = shallow(<Body {...props} />);
    // Assert
    expect(component).toMatchSnapshot();
  });
});
