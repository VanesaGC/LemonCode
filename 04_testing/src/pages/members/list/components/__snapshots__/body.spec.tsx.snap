// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagesmemberslistcomponents (body) tests should render as expected when is invoked 1`] = `
<tbody>
  <Row
    key="1"
    member={
      Object {
        "avatarUrl": "maria.png",
        "id": 1,
        "name": "María",
      }
    }
  />
  <Row
    key="2"
    member={
      Object {
        "avatarUrl": "pedro.png",
        "id": 2,
        "name": "Pedro",
      }
    }
  />
</tbody>
`;

exports[`pagesmemberslistcomponents (body) tests should render as expected when passing required properties 1`] = `
<tbody>
  <Row
    key="1"
    member={
      Object {
        "avatarUrl": "maria.png",
        "id": 1,
        "name": "María",
      }
    }
  />
  <Row
    key="2"
    member={
      Object {
        "avatarUrl": "pedro.png",
        "id": 2,
        "name": "Pedro",
      }
    }
  />
</tbody>
`;

exports[`pagesmemberslistcomponents tests should render as expected when is invoked 1`] = `
<tbody>
  <Component
    key="1"
    member={
      Object {
        "avatarUrl": "maria.png",
        "id": 1,
        "name": "María",
      }
    }
  />
  <Component
    key="2"
    member={
      Object {
        "avatarUrl": "pedro.png",
        "id": 2,
        "name": "Pedro",
      }
    }
  />
</tbody>
`;
